src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ModifiedControlVariableCheck.java

```diff
@@ -397,9 +393,7 @@ private void popCurrentVariables(int count) {
                 return MUTATION_OPERATIONS.contains(iteratingExpressionAST.getType());
             }).forEach(iteratingExpressionAST -> {
                 final DetailAST oneVariableOperatorChild = iteratingExpressionAST.getFirstChild();
-                if (oneVariableOperatorChild.getType() == TokenTypes.IDENT) {
```

Comment:
The only non-idents that showed up where `.` and you can't have a variable named that.